### PR TITLE
Add ProgressIndicator control

### PR DESCRIPTION
**Pull request by promag**
Add and use the `ProgressIndicator` control. It can be seen as a more light version of the `ProgressBar` control.

The example captures mouse clicks to update the progress.

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/Win64%20\[unit%20tests,%20no%20gui%20tests,%20no%20boost::process,%20no%20functional%20tests\]%20\[focal\]/insecure_win_gui.zip?branch=pull/69)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macOS%2010.15%20\[gui,%20no%20tests\]%20\[focal\]/insecure_mac_gui.zip?branch=pull/69)

### CODE REVIEW

**Comment by GBKS** [comment]
I am not able to build this one. Getting a series of `libbitcoin_util.a(libbitcoin_util_a-strnlen.o) has no symbols` messages. The main branch builds fine, what could be the problem here?

**Comment by promag** [comment]
Let me rebase this branch @GBKS

**Comment by hebasto** [comment]
@GBKS
> I am not able to build this one. Getting a series of `libbitcoin_util.a(libbitcoin_util_a-strnlen.o) has no symbols` messages.

These messages must be completely harmless.

**Comment by GBKS** [comment]
Thanks for looking into this. Here's more of the error message. Does not seem harmless, but I might not be interpreting this correctly.
```
make[3]: *** No rule to make target `libunivalue.la'.  Stop.
make[2]: *** [univalue/libunivalue.la] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

**Comment by promag** [comment]
Rebased.

**Comment by hebasto** [comment]
@GBKS 

> Thanks for looking into this. Here's more of the error message. Does not seem harmless, but I might not be interpreting this correctly.
> 
> ```
> make[3]: *** No rule to make target `libunivalue.la'.  Stop.
> make[2]: *** [univalue/libunivalue.la] Error 2
> make[1]: *** [all-recursive] Error 1
> make: *** [all-recursive] Error 1
> ```

What is your `./configure` srcript options?

File: src/qml/controls/ProgressIndicator.qml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    property real progress: 0
+    Behavior on progress {
+        SmoothedAnimation {
+            velocity: 1
+        }
+    }
+    contentItem: Rectangle {
+        implicitHeight: 6
+        radius: Math.floor(height / 2)
+        color: "#404040"
```
Comment (hebasto) [inline src/qml/controls/ProgressIndicator.qml:18]: Could we avoid magic numbers?

**Review by hebasto — COMMENTED** [review]
Tested 83c21f6f92a8b520bb722d820bc22e99de5cccc6 on Linux Mint 20.2 (Qt 5.12.8), and it looks great!

@promag 

Is it your intention to get the demo commit merged?

File: src/qml/controls/ProgressIndicator.qml (hunk 1 above)
Comment (promag) [reply src/qml/controls/ProgressIndicator.qml:18]: We need to move these constants somewhere, something to follow up.

**Comment by promag** [comment]
> Is it your intention to get the demo commit merged?

@hebasto yes I think it's fine. What would be the alternative? Drop the commit before merging? Or we should have a separate branch with these demos?

**Review by hebasto — APPROVED** [review]
ACK 83c21f6f92a8b520bb722d820bc22e99de5cccc6

**Comment by GBKS** [comment]
ACK. A small change would be to speed up the transition a little by setting velocity to 1.5.

Overall I'd prefer a consistent animation duration (200ms or so) for this component with a `cubic-bezier(0.5, 0.0, 0.2, 1)` easing curve, but we can tweak that some other time, not sure if QML is set up for that easily.

**Comment by promag** [comment]
@GBKS changed animation as suggested.

**Comment by Bosch-0** [comment]
Progress bar on Windows 10 is not full even though the network is up to sync. 

image

**Comment by GBKS** [comment]
ACK 1bf3d5d9287a0296b2d2af5cd690bab029f803f2. Thanks for the animation tweaks.

@Bosch-0 I think the bar is just meant as a sample UI element, and not to reflect the sync status.

**Comment by Bosch-0** [comment]
ACK https://github.com/bitcoin-core/gui-qml/commit/1bf3d5d9287a0296b2d2af5cd690bab029f803f2

Okay sweet, wasn't sure :)

**Pull request merged**